musmart/melody/contour/step_contour.py

```diff
@@ -0,0 +1,217 @@
+"""Calculates the Step Contour of a melody, along with related features, as implemented
+in the FANTASTIC toolbox of Müllensiefen (2009) [1].
+
+Examples
+--------
+>>> pitches = [60, 62, 64, 65, 67]  # C4, D4, E4, F4, G4
+>>> durations = [1.0, 1.0, 1.0, 1.0, 1.0]
+>>> sc = StepContour(pitches, durations)
+>>> sc.contour[:8]  # First 8 values of 64-length contour
+[60, 60, 60, 60, 60, 60, 60, 60]
+>>> sc.global_variation  # Standard deviation of contour
+2.3974...
+>>> sc.global_direction  # Correlation with ascending line
+0.9746...
+>>> sc.local_variation  # Average absolute difference between adjacent values
+0.1111...
+"""
+
+__author__ = "David Whyatt"
+
+import numpy as np
+
+
+class StepContour:
+    """Class for calculating and analyzing step contours of melodies.
```

Comment:
Does this analysis assume that the melody does not contain rests? We should say so.

Reply:
Rests are treated as extending the duration of the previous note - will add to docstring now